packages/plugin-ext-vscode/src/browser/plugin-vscode-commands-contribution.ts

```diff
@@ -330,6 +331,19 @@ export class PluginVscodeCommandsContribution implements CommandContribution {
             }
         });
 
+        commands.registerCommand({ id: 'workbench.action.revertAndCloseActiveEditor' }, {
+            execute: async () => {
+                const editor = this.editorManager.currentEditor;
+                if (editor) {
+                    const monacoEditor = MonacoEditor.getCurrent(this.editorManager);
+                    if (monacoEditor) {
+                        await monacoEditor.document.revert();
```

Comment:
Reference implementation: https://github.com/theia-ide/vscode/blob/d24b5f70c69b3e75cd10c6b5247a071265ccdd38/src/vs/workbench/browser/parts/editor/editorActions.ts#L538-L560

Important point that failed `revert` does not prevent closing the editor, they will still close it without saving. We need `try-catch` here. In catch block you can call `await ApplicationShell.closeWidget(editor.id, { save: false });`